pallets/messages/src/lib.rs

```diff
@@ -165,6 +167,50 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Adds a message for an offchain resource
+		/// # Arguments
+		/// * `origin` - A signed transaction origin from the provider
+		/// * `on_behalf_of` - Optional. The msa id of delegate.
+		/// * `schema_id` - Registered schema id for current message.
+		/// * `cid` - The content address for an IPFS payload
+		/// * `payload_length` - The size of the payload
+		/// * Returns
+		/// * [DispatchResultWithPostInfo](https://paritytech.github.io/substrate/master/frame_support/dispatch/type.DispatchResultWithPostInfo.html)
+		#[pallet::weight(T::WeightInfo::add_offchain(cid.get().len() as u32, 1_000))]
+		pub fn add_offchain(
+			origin: OriginFor<T>,
```

Comment:
Can we give it more descriptive name given it will be exposed to outside consumers

Like add_messages_offchain or add_messages_ipfs

Offchain could also mean offchain db so might create confusion

Reply:
Good point.